### PR TITLE
Allow more keys in graphql queries

### PR DESCRIPTION
**Pull request by christoph-maurer**
I have added `nearnormalHemisphericalVisibleReflectance`, `nearnormalHemisphericalSolarTransmittance`,
`nearnormalHemisphericalSolarReflectance` and `infraredEmittance` to
the API specification. If these values are contained explicitly in the resource of the optical data set, then programmers should also be able to use them in graphql queries. Examples of how these values can be represented in the resources can be found [here](https://github.com/ise621/building-envelope-data/issues/233#issuecomment-874690500).

I have added `cielabColor` and `colorRenderingIndex` to the API specification and rearrange them in the schemas.

I have also added the `formatId` of a `dataFormat` to the `DataPropositionInputs` so that it can be used in graphql queries.

Relates to #233 and ise621/metabase#72

### CODE REVIEW

File: apis/database.graphql [hunk 1]
```diff
@@ -570,12 +566,35 @@ input DataPropositionInput {
...
+  True for data for which at least one proposition in `or` is true, otherwise
+  false. In particular, it is false if there is no proposition in `or`.
+  """
+  or: [DataPropositionInput!]
+
+  """
+  True for data for the component with identifier `formatId`, otherwise
+  false.
+  """
+  formatId: UuidPropositionInput
```
Comment (simon-wacker) [inline apis/database.graphql:584]: Please use `dataFormatId` as field name (because it is not clear from the context that the format refers to the format of data). I would also change the description to:

> True for data with at least one resource for whose format identifier the proposition `dataFormatId` is true, otherwise false. See `GetHttpsResource#formatId` and `FileMetaInformation#formatId`.

File: apis/database.graphql [hunk 2]
```diff
@@ -607,9 +635,14 @@ See `DataPropositionInput`.
 input OpticalDataPropositionInput {
   componentId: UuidPropositionInput
   and: [OpticalDataPropositionInput!]
-  nearnormalHemisphericalVisibleTransmittance: FloatPropositionInput
   not: OpticalDataPropositionInput
   or: [OpticalDataPropositionInput!]
+  formatId: UuidPropositionInput
```
Comment (simon-wacker) [inline apis/database.graphql:640]: Rename field to `dataFormatId`.

File: apis/database.graphql [hunk 3]
```diff
@@ -618,9 +651,10 @@ See `DataPropositionInput`.
 input CalorimetricDataPropositionInput {
   componentId: UuidPropositionInput
   and: [CalorimetricDataPropositionInput!]
-  gValue: FloatPropositionInput
   not: CalorimetricDataPropositionInput
   or: [CalorimetricDataPropositionInput!]
+  formatId: UuidPropositionInput
```
Comment (simon-wacker) [inline apis/database.graphql:656]: Rename field to `dataFormatId`.

File: apis/database.graphql [hunk 4]
```diff
@@ -632,6 +666,7 @@ input HygrothermalDataPropositionInput {
   and: [HygrothermalDataPropositionInput!]
   not: HygrothermalDataPropositionInput
   or: [HygrothermalDataPropositionInput!]
+  formatId: UuidPropositionInput
```
Comment (simon-wacker) [inline apis/database.graphql:669]: Rename field to `dataFormatId`.

File: apis/database.graphql [hunk 5]
```diff
@@ -642,6 +677,7 @@ input PhotovoltaicDataPropositionInput {
   and: [PhotovoltaicDataPropositionInput!]
   not: PhotovoltaicDataPropositionInput
   or: [PhotovoltaicDataPropositionInput!]
+  formatId: UuidPropositionInput
```
Comment (simon-wacker) [inline apis/database.graphql:680]: Rename field to `dataFormatId`.

File: apis/database.graphql [hunk 6]
```diff
@@ -918,10 +954,28 @@ type OpticalData implements Node & Data {
   approval: ResponseApproval!
 
   """
-  Mirrored nearnormal hemispherical visible transmittance values that occur in
-  the data
+  Mirrored nearnormal hemispherical visible transmittance values which can be found in the data set. If it can only be calculated with the data set, then nearnormalHemisphericalVisibleTransmittances must not be returned.
```
Comment (simon-wacker) [inline apis/database.graphql:957]: Here and below I would leave `that occur in the data` instead of changing it to `which can be found in the data set`.

Comment (simon-wacker) [inline apis/database.graphql:957]: Here and below I would rephrase `If it can only be calculated with the data set, then nearnormalHemisphericalVisibleTransmittances must not be returned.` to `If some or all of those values do not occur verbatim in the data, they need not occur in the list, even if they can be calculated based on the data.`

**Review by simon-wacker — CHANGES_REQUESTED** [review]
After incorporating and committing the proposed changes please run `make format` within a running Docker container (the shell in which you also run `make test` etc.) and commit the changes made by running this command.

**Review by simon-wacker — CHANGES_REQUESTED** [review]
If we are already at it, we should also add

```
type CieLabColor {
  lStar: Float!
  aStar: Float!
  bStar: Float!
}
```

now. This makes it more likely that some databases support those in the future. Note that I used upper camelcase for the type name and lower camelcase for all field names because that is the standard in GraphQL and everyone would get confused if that is different (in particular, automatic mappers from GraphQL to C# or Python types). The documentation of those fields should mention `L*`, `a*`, and `b*` with their usual case.

I also did not use `type CieLabParameterSet` because it is not a `set` but a product type and this information is already in `type` (so no need to have that in the name) and I also used `Color` instead of `Parameters` because I think that makes the intend of the type clearer. If you don't think that `Color` is correct than I would use `Coordinates` because that is how [Wikipedia (and hopefully also the official documentation) refers to the three values](https://en.wikipedia.org/wiki/CIELAB_color_space#CIELAB_coordinates).

**Review by simon-wacker — CHANGES_REQUESTED** [review]
If we are already at it, we should also add color rendering index. I have no proposal for its exact representation because I do not know enough about it. Can you make a proposal?

File: apis/database.graphql (hunk 1 above)
Comment (christoph-maurer) [reply apis/database.graphql]: In 263e8c3fa8a3f645cdb62a0344078da7f0969e0b , I have substituted all `formatId` by `dataFormatId`. Or should we continue to use `formatId` for `GetHttpsResource` and `FileMetaInformation`?

Comment (simon-wacker) [reply apis/database.graphql]: Ah, yes, renaming all fields seems like a good idea. The latter two are breaking changes though and need to be communicated with Stephen so that he adapts his implementation.

Comment (christoph-maurer) [reply apis/database.graphql]: @StephenCzarnecki We plan a breaking change to rename `formatId` to `dataFormatId`. Is this ok for you?

Comment (StephenCzarnecki) [reply apis/database.graphql]: @christoph-maurer Yes, that is fine.  It is an easy thing to change.

**Comment by christoph-maurer** [comment]
@simon-wacker Currently, `color` is used in `component/characteristics` and in `photovoltaics` for the subcomponents. Is this the best solution or would you prefer to have [color](https://github.com/ise621/building-envelope-data/blob/615f3e1f6dfab56c25162f6e12a1a11b3b3cf0a3/schemas/common.json#L418) as an additional subschema of `opticalData`? If `color` remains a property of `component/characteristics`, I would implement it in graphql accordingly.

**Comment by christoph-maurer** [comment]
I have discussed the changes with @simon-wacker and I have implemented them accordingly.

**Comment by christoph-maurer** [comment]
I've closed the pull request by mistake.

File: apis/database.graphql [hunk 7]
```diff
@@ -918,10 +1006,38 @@ type OpticalData implements Node & Data {
...
+
+  """
+  Mirrored infrared emittance values that occur in the data. If some or all of those values do not occur verbatim in the data, they need not occur in the list, even if they can be calculated based on the data.
+  """
+  infraredEmittances: [Float!]!
+
+  """
+  Mirrored values for the Color Rendering Index that occur in the data. If some or all of those values do not occur verbatim in the data, they need not occur in the list, even if they can be calculated based on the data.
+  """
+  colorRenderingIndexes: [Float!]!
```
Comment (simon-wacker) [inline apis/database.graphql:1035]: In technical writing `indices` is the preferred spelling: https://english.stackexchange.com/questions/61080/the-plural-of-index-indexes-or-indices/3126#3126

**Comment by christoph-maurer** [comment]
> Why did you move the definition of `color` from `common.json` to `opticalData.json` when the definition is also used and needed in `photovoltaicData.json`?

From my point of view, `color` is now part of the domain `optical`. It is still used in `photovoltaicData.json` to define the colors of subcomponents which may not have their own identifier. But the color of the module is now defined in `opticalData`.

File: apis/database.graphql (hunk 7 above)
Comment (christoph-maurer) [reply apis/database.graphql]: Done with d277de5
